docs/features/software-catalog/api.md

```diff
@@ -542,7 +608,219 @@ If the location already exists the response will be `HTTP/1.1 409 Conflict` and
 
 Supports the `?dryRun=true` query parameter, which will perform validation and not write anything to the database. In the event of successfully passing validation, the `entities` field of the response JSON will be populated with entities present in the location.
 
-### `DELETE /locations/<uid>`
+### `POST /analyze-location`
+
+Validate a given location.
+
+Request body is JSON, on the form
+
+```json
+{
+  "location": {
+    "type": "<string>",
+    "target": "<string>"
+  },
+  "catalogFileName": "<string>"
+}
+```
+
+And Response type is JSON, on the form
+
+```json
+{
+  "generateEntities": [
+    {
+      "fields": [
+        {
+          "description": "<string>",
+          "value": "<string>",
+          "state": "needsUserInput",
+          "field": "<string>"
+        },
+        {
+          "description": "<string>",
+          "value": "<string>",
+          "state": "analysisSuggestedNoValue",
+          "field": "<string>"
+        }
+      ],
+      "entity": {
```

Comment:
Does it make sense to list all the entity properties here? Would it be better to just note it's an Entity object?

Reply:
Yes. It doesn not. Removed useless lines.